### PR TITLE
fix: Trim matched quotes from ovf properties

### LINKED ISSUE #824: [BUG] Proxy Settings stop installation
**Describe the bug**
When setting the proxy during ovf install the description of the proxy setting states it can be set with Quotes.
Unfortunately the installscript does not like this.
If you configure the Proxy like this : "http://proxy:port/"
the deployment script hangs and veba is not beeing installed.
this is because the Quotes close some variabe name in the installscript and you get a error.

**To Reproduce**
1.deploy the ovf 
2. In the Wizzard set a Proxy Server according to description.
3. Finish deployment wizard
4. Turn on VM and have a look at the console.

**Expected behavior**
The proxy is set corectly and kubernetes is beeing deployed.

**Version:**
 - VEBA Form Factor: Appliance
 - VEBA Version: 0.7.2

**Additional context**
There should be a check to accept both versions with and without quotes 
to clean the input before usage as variable in the script.

### PR DESCRIPTION
**Pull request by dsbibby**
Signed-off-by: David Bibby <davidsbibby@gmail.com>

## Summary

Removed quotes from proxy config examples.
Added line to strip matched pairs of single or double quotes enclosing any ovf property value in `getOvfProperty.py`
Added `test_getOvfProperty.py` with pytest unit tests.
Some code style changes and small bugs fixed in `getOvfProperty.py`

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [x] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [x] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [x] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [x] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [x] Please include any relevant screenshots and/or output as part of your testing
- [x] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation changes
- [ ] Other (please describe):

## Resolved Issues

List of Issues closed or resolved by this PR. Add multiple `Closes` keyword followed by the issue number (e.g. Closes #ISSUE-NUMBER)

Closes #824 

## Testing Verification

* Unit tests written and all passing
* Built and deployed appliance, using quoted proxy config

chrome_iTHZnz3xsI

Proxy config entered with quotes
chrome_yjTdXmiRsc

Build proceeds OK
chrome_K8yNpCm4ny

Config in appliance has extra quotes removed
Tabby_979o86v4T3

## Additional Information

* Any other details you wish to include or mention

If you have any questions/comments, feel free to reach out to team on Slack [#vcenter-event-broker-appliance](https://vmwarecode.slack.com/archives/CQLT9B5AA)

Thank you from the VEBA Team! 🥳

### CODE REVIEW

**Comment by embano1** [comment]
Awesome! Can you please also update the commit title/message as per my description in your other PR?

File: files/getOvfProperty.py [hunk 1]
```diff
@@ -35,18 +38,20 @@ def main(argv):
...
+
     try:
         res = ovf[argv[0]]
     except KeyError as err:
         debug(f'ovfProperty not found: {err}')
         sys.exit(1)
-    
+
+    # Strip enclosing quotes
+    res = re.sub(r'^(["\'])(.*)\1$', r'\2', res)
```
Comment (embano1) [inline files/getOvfProperty.py:51]: question: this will only match enclosing (begin/end) quotes (" or ') but not if they'd appear within the string?

**Comment by embano1** [comment]
Did you also attempt (validate) to build an appliance with the fixes and a valid property file?

File: files/getOvfProperty.py (hunk 1 above)
Comment (dsbibby) [reply files/getOvfProperty.py:51]: That's correct - there are unit tests that cover the following scenarios:
- `"Hello World"` -> quotes are stripped
- `'Hello World'` -> quotes are striped
- `'Hello World"` -> quotes are not stripped
- `Hello World"` -> quotes are not stripped
- `& < > ' "` -> quotes are not stripped

However, having slept on it I realise there is an edge case where a password could legitimately have a leading and trailing quote. Passwords as a rule should pass through unaltered, so I'll update the PR with a suggestion as to how to handle this.

**Comment by dsbibby** [comment]
> Did you also attempt (validate) to build an appliance with the fixes and a valid property file?

When you say "valid property file", do you mean proxy values without quotes? I did not (but I will as part of the update I'll add for the above), however all property values are affected by this change so in effect all other property values that were passed in my original test (IPs, hostnames, etc.) tested this. Regardless I'll update the deployment test with quoted and unquoted proxy values.

File: files/getOvfProperty.py [hunk 2]
```diff
@@ -47,8 +47,10 @@ def main(argv):
         debug(f'ovfProperty not found: {err}')
         sys.exit(1)
 
-    # Strip enclosing quotes
-    res = re.sub(r'^(["\'])(.*)\1$', r'\2', res)
+    # Strip enclosing quotes if not a password
+    if not re.search('password', argv[0], flags=re.IGNORECASE):
```
Comment (dsbibby) [inline files/getOvfProperty.py:51]: This is the only way I could think to detect passwords - it assumes that the OVF Property for a password will have the phrase `password` in its name.

**Comment by dsbibby** [comment]
Hi @embano1,

Apologies this took a while longer than expected. Had an awful fight getting quote escaping working in all the right places.

Let me know your thoughts and if any other changes are needed.

David

**Comment by embano1** [comment]
This looks pretty solid IMHO! Since we're doing pre-release builds as part of our next release, we'd detect breaking changes by this. But I don't expect them because you have great test coverage.

Can you please create a final (squashed) commit?

**Review by embano1 — APPROVED** [review]
THX A TON!

**Pull request merged**